R/epi_recipe.R

```diff
@@ -276,13 +280,117 @@ is_epi_recipe <- function(x) {
+#'
+#' workflow
+#'
 add_epi_recipe <- function(
     x, recipe, ..., blueprint = default_epi_recipe_blueprint()) {
   workflows::add_recipe(x, recipe, ..., blueprint = blueprint)
 }
 
+#' @rdname add_epi_recipe
+#' @export
+remove_epi_recipe <- function(x) {
+  workflows:::validate_is_workflow(x)
+
+  if (!workflows:::has_preprocessor_recipe(x)) {
+    rlang::warn("The workflow has no recipe preprocessor to remove.")
+  }
 
+  actions <- x$pre$actions
+  actions[["recipe"]] <- NULL
 
+  new_epi_workflow(
+    pre = workflows:::new_stage_pre(actions = actions),
+    fit = x$fit,
+    post = x$post,
+    trained = FALSE
+  )
+}
+
+#' @rdname add_epi_recipe
+#' @export
+update_epi_recipe <- function(x, recipe, ..., blueprint = NULL) {
+  x <- remove_epi_recipe(x)
+  add_epi_recipe(x, recipe, blueprint = blueprint)
+}
+
+#' Adjust a step in an `epi_workflow` or `epi_recipe`
+#'
+#' Make a parameter adjustment to a step in either an
+#' `epi_workflow` or `epi_recipe` object.
+#'
+#'
+#' @details This function can either adjust a step in a `epi_recipe` object
+#' or a step from a `epi_recipe` object in an `epi_workflow`. In any case, the
+#' argument name and update value must be inputted as `...`.
+#' See the examples below for brief illustrations of both types of updates.
+#'
+#' @param x A `epi_workflow` or `epi_recipe` object
+#'
+#' @param step_num the number of the step to adjust
+#'
+#' @param ... Used to input a parameter adjustment
+#'
+#' @param blueprint A hardhat blueprint used for fine tuning the preprocessing.
+#'
+#' @return
+#' `x`, updated with the adjustment to the specified `epi_recipe` step.
+#'
+#' @export
+#' @examples
+#' jhu <- case_death_rate_subset %>%
+#'   filter(time_value > "2021-11-01", geo_value %in% c("ak", "ca", "ny"))
+#' r <- epi_recipe(jhu) %>%
+#'   step_epi_lag(death_rate, lag = c(0, 7, 14)) %>%
+#'   step_epi_ahead(death_rate, ahead = 7) %>%
+#'   step_epi_naomit()
+#'
+#' wf <- epi_workflow(r, parsnip::linear_reg()) %>% fit(jhu)
+#' latest <- jhu %>%
+#'   dplyr::filter(time_value >= max(time_value) - 14)
+#'
+#' # Adjust `step_epi_ahead` to have an ahead value of 14
+#' # in the `epi_workflow`
+#' wf2 = wf %>% adjust_epi_recipe(step_num = 2, ahead = 14)
+#' workflows::extract_preprocessor(wf2)
+#'
+#' # Adjust `step_epi_ahead` to have an ahead value of 14
+#' # in the `epi_recipe`
+#' r2 = r %>% adjust_epi_recipe(step_num = 2, ahead = 14)
+#' r2
+#'
+adjust_epi_recipe <- function(x, step_num, ..., blueprint = default_epi_recipe_blueprint()) {
+  UseMethod("adjust_epi_recipe")
+}
+
+#' @rdname adjust_epi_recipe
+#' @export
+adjust_epi_recipe.epi_workflow <- function(
+    x, step_num, ..., blueprint = default_epi_recipe_blueprint()) {
+
+  recipe <- workflows::extract_preprocessor(x)
+  recipe$steps[[step_num]] <- update(recipe$steps[[step_num]], ...)
+
+  update_epi_recipe(x, recipe, blueprint = blueprint)
+}
+
+#' @rdname adjust_epi_recipe
+#' @export
+adjust_epi_recipe.epi_recipe <- function(
```

Comment:
I think I've mentioned this, but I'd like to see this allow for `step_name` as well. You can access these with something like:
`map_chr(x$steps, ~ attr(.x, "class")[1])`.

Then if it's a unique match, use it. Otherwise, return an error that gives the numbers that would have been available.

Reply:
Also worth mentioning that one can use `tidy(recipe)` to see some summary information about the recipe. We should make a similar method for the frosting. And provide some documentation (short Vignette?) that illustrates all of these as well as updating.